modules/webgl/src/adapter/resources/webgl-render-pipeline.ts

```diff
@@ -393,6 +393,7 @@ export class WEBGLRenderPipeline extends RenderPipeline {
         continue; // eslint-disable-line no-continue
       }
       this.vertexArrayObject.setConstant(attributeInfo.location, value);
+      this.vertexArrayObject.enable(attributeInfo.location, false);
```

Comment:
Shouldn't this go into `VertexArrayObject.setConstant`? `enable` is called in `VertexArrayObject.setBuffer`

Reply:
Yes I was considering it. One small hesitation was that in principle, the vertexArray does not have to be updated every draw, as each model could have its own vertexArray and just bind it, even when they share programs. 

If we did that, it would mean that the location disable state could also be saved, however the constant values must still be updated on every draw as they are stored on the shared context and might have been clobbered by the constant values of other draw calls.

So there is perhaps a small case for keeping these separate. Will give it some more thought.